CRM/Speakcivi/Logic/Campaign.php

```diff
@@ -486,22 +486,27 @@ public function isYoumove() {
    * @throws \CRM_Speakcivi_Exception
    */
   public function getContent($url, $authString = NULL) {
-    $ch = curl_init();
-    curl_setopt($ch, CURLOPT_URL, $url);
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
-    if ($authString != NULL) {
-      curl_setopt($ch, CURLOPT_USERPWD, $authString);
-    }
-    $data = curl_exec($ch);
-    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-    curl_close($ch);
-    if ($code == 200) {
-      return $data;
-    } elseif ($code = 404) {
-      throw new CRM_Speakcivi_Exception('Speakout campaign doesnt exist: ' . $url, 1);
-    } else {
-      throw new CRM_Speakcivi_Exception('Speakout campaign is unavailable' . $url, 2);
-    }
+    return fetchSpeakoutContent($url, $authstring = NULL);
+
+  }
+}
+
+function fetchSpeakoutContent($url, $authString = NULL) {
```

Comment:
I see this method *fetchSpeakoutContent* is used by two class, CRM_Speakcivi_Logic_Campaign and CRM_Speakcivi_Logic_Cache_SpeakoutCampaign.

It looks bad as a function outside any class. I think it should be in some of those class, for example in CRM_Speakcivi_Logic_Cache_SpeakoutCampaign as static method.